### ISSUE TITLE
Tool to find existing duplicate

### ISSUE THREAD

**Issue by jpyrat** [shaarli/Shaarli#1303]
Hi,

I have duplicates in my base (eg : https://veille.pyrat.net/?searchterm=moving+to+https&searchtags=wordpress)

Could be usefull to have a tool to find duplicate and delete them

**Comment by nodiscc** [comment]
@jpyrat Hi, normally when posting a link, if the URL is already in the database, the [_Edit Shaare_ dialog is whon](https://github.com/shaarli/Shaarli/blob/786f35f2700d83687ef494b98e3dace37d268e1c/index.php#L1345) instead of the _Add Shaare_ dialog. Nothing fundamentally prevents 2 shaares with the same URL to be present in the datastore, but posting links from the web interface should prevent such duplication (except some edge cases, like 2 _Add Shaare_ dialogs opened at the same time).

How did you post these links? API? Third-party tool?

**Comment by ArthurHoaro** [comment]
@nodiscc Yes it'll be the oldest one, as links are ordered by dates descending, and the URL array is loaded in in order. However in this example all links have the same date.

This is an edge case which is not supposed to happen ; also your example is quite old. So, IMO, there shouldn't be a core feature trying to fix it. Either there is a bug somewhere that allowed this to happen (REST API maybe?) that needs to be fixed, or either it's just something that you could fix manually/using a custom script.

**Comment by kalvn** [comment]
Another idea would be to have a way to find links which are very likely the same but not exactly. For example a link that you previously saved in HTTP and that you save again in HTTPS. Or similar links but with close but different query string parameters. But I guess this is not easy to do.

**Comment by TeoPetterson** [comment]
> Hi,
> 
> I have duplicates in my base (eg : https://veille.pyrat.net/?searchterm=moving+to+https&searchtags=wordpress)
> 
> Could be usefull to have a tool to find duplicate and delete them

Hello sir,
how did you post in this website ?  (https://veille.pyrat.net/?searchterm=moving+to+https&searchtags=wordpress) I'm insterested about that.

**Comment by dsalo** [comment]
This can also be an artifact of importing from another bookmarking tool into Shaarli. (Speaking from experience!) If it's not too much hassle to build a duplicate-detector, I'd definitely find utility in it.
